File: signalpilot/web/components/notebook/analysis-trails.ts
export const ANALYSIS_BRANCH_PREFIX = "analysis/";
export const ANALYSIS_NOTEBOOK_RELPATH_PREFIXES = {
  notion: "notebooks/notion/",
  slack: "notebooks/slack/",
} as const;

export function isGeneratedAnalysisTrailNotebook({
  project,
  branch,
  file,
}: {
  project?: string | null;
  branch?: string | null;
  file?: string | null;
}): boolean {
  if (!project || !branch || !file) return false;
  if (!branch.startsWith(ANALYSIS_BRANCH_PREFIX)) return false;

  const normalized = file.replace(/\\/g, "/").replace(/^\/+/, "");
  return Object.entries(ANALYSIS_NOTEBOOK_RELPATH_PREFIXES).some(
    ([source, prefix]) =>
      branch.startsWith(`${ANALYSIS_BRANCH_PREFIX}${source}/`) &&
      normalized.startsWith(prefix),
  );
}

export const ANALYSIS_TRAIL_CONFIG_OVERRIDES = {
  runtime: {
    auto_instantiate: false,
  },
} as const;

File: signalpilot/notebook-server/signalpilot/_server/api/endpoints/ws/analysis_trails.py
from __future__ import annotations

ANALYSIS_BRANCH_PREFIX = "analysis/"
ANALYSIS_NOTEBOOK_RELPATH_PREFIXES = {
    "notion": "notebooks/notion/",
    "slack": "notebooks/slack/",
}


def is_generated_analysis_trail_notebook(
    *,
    project_id: str | None,
    branch: str | None,
    file_key: str | None,
) -> bool:
    """Return True for project-backed generated Slack/Notion analysis trails."""
    if not project_id or not branch or not file_key:
        return False
    if not branch.startswith(ANALYSIS_BRANCH_PREFIX):
        return False

    normalized = file_key.replace("\\", "/").lstrip("/")
    return any(
        branch.startswith(f"{ANALYSIS_BRANCH_PREFIX}{source}/")
        and normalized.startswith(prefix)
        for source, prefix in ANALYSIS_NOTEBOOK_RELPATH_PREFIXES.items()
    )
